packages/records/lib/models/records.integration.test.ts

```diff
@@ -211,6 +211,55 @@ describe('Records service', () => {
     });
 
     describe('updating records', () => {
+        it('should deep merge records', async () => {
+            const connectionId = rnd.number();
+            const environmentId = rnd.number();
+            const model = rnd.string();
+            const syncId = uuid.v4();
+            const records = [{ id: '1', person: { name: 'John Doe', age: 35, children: [{ name: 'Jenny Doe', age: 3 }] } }];
+
+            const inserted = await upsertRecords({ records, connectionId, environmentId, model, syncId, syncJobId: 1 });
+            expect(inserted).toStrictEqual({
+                addedKeys: ['1'],
+                updatedKeys: [],
+                deletedKeys: [],
+                nonUniqueKeys: [],
+                nextMerging: {
+                    strategy: 'override'
+                }
+            });
+
+            const updated = await updateRecords({
+                records: [{ id: '1', person: { age: 36, children: [{}, { name: 'Maurice Doe', age: 1 }] } }],
+                connectionId,
+                model,
+                syncId,
+                syncJobId: 2
+            });
+            expect(updated).toStrictEqual({
+                addedKeys: [],
+                updatedKeys: ['1'],
+                deletedKeys: [],
+                nonUniqueKeys: [],
+                nextMerging: {
+                    strategy: 'override'
+                }
+            });
+
+            const { records: found } = (await Records.getRecords({ connectionId, model })).unwrap();
+            expect(found.length).toBe(1);
+            expect(found?.[0]).toMatchObject({
+                person: {
+                    name: 'John Doe',
+                    age: 36,
+                    children: [
+                        { name: 'Jenny Doe', age: 3 },
+                        { name: 'Maurice Doe', age: 1 }
+                    ]
+                }
+            });
+        });
```

Comment:
Would love more test:
- where you swap array items
- where you add an item 
- where you remove an item

Reply:
:+1: I'll probably extract the merge code to another function and unit test that, just so it's not so heavy a test suite there, if that's cool.